src/lib/utils/thorchain/lp.ts

```diff
@@ -0,0 +1,83 @@
+import { type AccountId, type AssetId, fromAccountId, fromAssetId } from '@shapeshiftoss/caip'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { assetIdToPoolAssetId } from 'lib/swapper/swappers/ThorchainSwapper/utils/poolAssetHelpers/poolAssetHelpers'
+import { isUtxoChainId } from 'state/slices/portfolioSlice/utils'
+
+import { getAccountAddresses } from '.'
+import type {
+  MidgardLiquidityProvider,
+  MidgardPool,
+  ThorchainLiquidityProvidersResponseSuccess,
+  ThorNodeLiquidityProvider,
+} from './lp/types'
+
+export const getAllThorchainLiquidityProviderPositions = async (
+  assetId: AssetId,
+): Promise<ThorchainLiquidityProvidersResponseSuccess> => {
+  const poolAssetId = assetIdToPoolAssetId({ assetId })
+
+  if (!poolAssetId) throw new Error(`Pool asset not found for assetId ${assetId}`)
+
+  const { data } = await axios.get<ThorchainLiquidityProvidersResponseSuccess>(
+    `${
+      getConfig().REACT_APP_THORCHAIN_NODE_URL
+    }/lcd/thorchain/pool/${poolAssetId}/liquidity_providers`,
+  )
+
+  if (!data || 'error' in data) return []
+
+  return data
+}
+
+export const getThorchainLiquidityProviderPosition = async ({
+  accountId,
+  assetId,
+}: {
+  accountId: AccountId
+  assetId: AssetId
+}): Promise<(ThorNodeLiquidityProvider & MidgardPool) | null> => {
+  const poolAssetId = assetIdToPoolAssetId({ assetId })
+
+  const accountPosition = await (async () => {
+    const address = fromAccountId(accountId).account
+    if (!isUtxoChainId(fromAssetId(assetId).chainId))
+      return (
+        await axios.get<ThorNodeLiquidityProvider>(
+          `${
+            getConfig().REACT_APP_THORCHAIN_NODE_URL
+          }/lcd/thorchain/pool/${poolAssetId}/liquidity_provider/${address}`,
+        )
+      ).data
+
+    const liquidityProviderPositionsResponse =
+      await getAllThorchainLiquidityProviderPositions(assetId)
+
+    const allPositions = liquidityProviderPositionsResponse
+    if (!allPositions.length) {
+      throw new Error(`No LP positions found for asset ID: ${assetId}`)
+    }
+
+    const accountAddresses = await getAccountAddresses(accountId)
+
+    return allPositions.find(position => accountAddresses.includes(position?.asset_address ?? ''))
+  })()
+  if (!accountPosition) return null
+
+  // TODO(gomes): asset_address *or* rune_address when implementing sim. pools
+  const { data: midgardLiquidityProvider } = await axios.get<MidgardLiquidityProvider>(
+    `${getConfig().REACT_APP_MIDGARD_URL}/member/${accountPosition.asset_address}`,
```

Comment:
We'll need to checksum `asset_address`.

✅ https://indexer.thorchain.shapeshift.com/v2/member/0x32DBc9Cf9E8FbCebE1e0a2ecF05Ed86Ca3096Cb6 (manually constructed)

❌ https://indexer.thorchain.shapeshift.com/v2/member/0x32dbc9cf9e8fbcebe1e0a2ecf05ed86ca3096cb6 (this PR)

Reply:
That's an odd one!
Was testing https://indexer.thorchain.shapeshift.com/v2/member/0x5daf465a9ccf64deb146eeae9e7bd40d6761c986 and it worked fine without chekcsum. In https://indexer.thorchain.shapeshift.com/v2/members your address isn't checksummed iether, but for some reason it looks like some addresses *may* require checksum so will implement it across the board

Reply:
Also tackled in the wire-up, gm! 68127fb39a9cd57e24fe8dc460fc2371527c1827
<img width="1353" alt="image" src="https://github.com/shapeshift/web/assets/17035424/deadf55f-b46d-4a96-b266-62e825a9b0dc">